R/finalize_explanation.R

```diff
@@ -24,20 +24,35 @@ finalize_explanation <- function(vS_list, internal) {
 
   # internal$timing$shapley_computation <- Sys.time()
 
-
-  # Clearnig out the tmp list with model and predict_model (only added for AICc-types of empirical approach)
+  # Clearing out the tmp list with model and predict_model (only added for AICc-types of empirical approach)
   internal$tmp <- NULL
 
   internal$output <- processed_vS_list
 
-
   output <- list(
     shapley_values = dt_shapley,
     internal = internal,
     pred_explain = p
   )
   attr(output, "class") <- c("shapr", "list")
 
+  # Compute the MSEv evaluation criterion if the output of the predictive model is a scalar.
+  # TODO: check if it makes sense for output_size > 1.
+  if (internal$parameters$output_size == 1) {
+    output$MSEv_eval_crit <- compute_MSEv_eval_crit(
+      internal = internal,
+      processed_vS_list = processed_vS_list,
+      p = p,
+      exclude_empty_grand_coalitions = ifelse(!is.null(internal$parameters$exclude_empty_grand_coalitions),
+                                              internal$parameters$exclude_empty_grand_coalitions,
+                                              FALSE),
+      use_shapley_weights = ifelse(!is.null(internal$parameters$use_shapley_weights),
+                                   internal$parameters$use_shapley_weights,
+                                   FALSE),
```

Comment:
- parameters in internal should be declared in the top of the function for clarity and readability on what is used 
	within the function. Just write e.g. aa = internal$parameters$blabla$aa on the top and use just aa later. 
- consider changing name on exclude_empty_grand_coalitions and use_shapley weights
- where are these two parameters defined? are they added to the output of explain()? I dont think so

Reply:
- In the new version I have declared the parameters at the begining.
- I have changed the names `exclude_empty_grand_coalitions` and `use_shapley weights` to `MSEv_skip_empty_full_comb` and `MSEv_uniform_comb_weights`, respectively. 
- They were poorly defined in the sense that the user could send them as `...` parameters in `explain()`. If they did not, then we used it as specified above. They were added to the output of `explain()`, IF the user specified them. I have now made them as arguments to `explain()`, made a proper test for valid input in `setup()` and added documentation to `explain()`.